### ISSUE TITLE
False positive prefer-angular-composition false positive with unsubscribe in super class

### ISSUE THREAD

**Issue by apeeters** [cartant/rxjs-tslint-rules#119]
The following code snippet does proper unsubscribe handling in ngOnDestroy but results in a false positive lint error.

```
@Component({
  selector: "superclass-composed-component",
  template: "<span>{{ value }}</span>"
})
export class SuperclassComposedComponent extends UnsubscribeOnDestroy implements OnInit {
  value: string;
  ngOnInit() {
    this.subscription.add(of("foo").subscribe(value => this.value = value));
  }
}

abstract class UnsubscribeOnDestroy implements OnDestroy {
  subscription = new Subscription();
  ngOnDestroy() {
    this.subscription.unsubscribe();
  }
}
```

**Comment by cartant** [comment]
Same response as here: https://github.com/cartant/rxjs-tslint-rules/issues/113#issuecomment-578473623
